extensions/http-curl/sitetosite/HTTPProtocol.cpp

```diff
@@ -132,10 +132,10 @@ int HttpSiteToSiteClient::readResponse(const std::shared_ptr<Transaction> &trans
           logger_->log_debug("confirm read for %s, but not finished ", transaction->getUUIDStr());
           if (stream->waitForDataAvailable()) {
             code = CONTINUE_TRANSACTION;
+            return 1;
           }
         }
 
-        closeTransaction(transaction->getUUIDStr());
```

Comment:
What error did this cause? It really seems strange to close the transaction if there is still data to read, but I am not sure what the exact consequences were.

Reply:
In this case the transaction as closed and recreated without any need of doing so. Not great, not terrible.